src/plugins/kibana/server/routes/api/ingest/register_simulate.js

```diff
@@ -0,0 +1,25 @@
+const _ = require('lodash');
+const buildRequest = require('../../../lib/ingest_build_request');
+const processResponse = require('../../../lib/ingest_process_response');
+
+module.exports = function registerSimulate(server) {
+  server.route({
+    path: '/api/kibana/ingest/simulate',
```

Comment:
This could cause a conflict if an index pattern happened to have the ID `simulate`. It also feels a bit weird to put it under ingest because that endpoint also creates an index pattern and template. Thoughts on `/api/kibana/simulate/pipeline` or `/api/kibana/pipeline/simulate` or something else?